bamboo/abort-if-not-pr-or-master.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+source .bamboo_env_vars || true
```

Comment:
Could we still run a manual build that does not have a PR open?

Reply:
@laurenfrederick  yes, you'd go to the branch, run a customized plan and override the GIT_PR variable.   Due to the way the `set-bamboo-env-variables.sh` script was written, if `GIT_PR` is set in the environment, it will never update the value in all subsequent stages/jobs